### ISSUE TITLE
Export as image support

### ISSUE THREAD

**Issue by zellerin** [minad/osm#9]
Hi,

are there any plans to support exporting osm links in org mode as an image? (e.g., as tile links for html or TeX - pins might be more complicated). 

If not planned by you, would patch be eventually considered (with or without paperwork)?

**Comment by minad** [comment]
It would be nice to have image export as a separate command and also for org links. I would accept a PR which adds this, it requires the paperwork though. Ideally a self-contained SVG is generated with tiles embedded via base64 encoding. This should be easy to implement based on the existing display code.

**Comment by zellerin** [comment]
Something like #13 for export? This is indeed easy, but I am not sure whether base64 is best way to do it, or links (to cache? to a server?) would be better.

The org-export would be bit more complicated - the tiles need not be in the cache, and it is not clear what should be default width and height.

**Comment by minad** [comment]
Thanks. Can you reuse more of the display code? I would also like to see pin and track export and ideally we don't duplicate the code. Base64 is okay since this gives us a self contained SVG. There is no other way. I want to have a self contained image in any case.

**Comment by zellerin** [comment]
Pins are exported, without much code duplication, but the substring based extraction from svg (to add x and y) is ugly. Cleaner code and reuse will need a bit more refactoring of the original code.

**Comment by minad** [comment]
Sure you can try to refactor some things. The display code would benefit from some decomposition in smaller functions. In your PR currently the svg tiles seem to be exported as links on Emacs 28? Instead all tiles should be base64 encoded.

**Comment by minad** [comment]
Also the question is if the reuse on the level of the tiles is the right approach and if it wouldn't be better to draw the overlays only once on top. Otherwise all the overlays are duplicated for neighboring tiles. I think it is better to use tiles for the display, but for the export we could do better. I also considered to use one big svg for the display but I think this would hurt performance. But I haven't tried. Maybe it is still good enough and one could aggressively clean the image cache with `clear-image-cache` when generating a new svg.
